packages/workbox-build/src/lib/errors.js

```diff
@@ -104,4 +104,10 @@ module.exports = {
   'bad-manifest-transforms-return-value': ol`The return value from a
     manifestTransform should be an object with 'manifest' and optionally
     'warnings' properties.`,
+  'string-entry-warning': ol`Some URLs were passed to additionalManifestEntries
```

Comment:
I'm interested in feedback here: my thinking is that we can support passing in a string URL, but if someone does that, we should let them know about the potential pitfalls involved with precaching a URL that may or may not be revisioned. They can avoid the warning by explicitly using `{url: '...'}` as the entry, kind of as a "I know what I'm doing and accept the risks mode."

The thing a like about that approach is that the most likely think developers will do first is pass in a string, and they'll see this warning, and hopefully they'll think about whether it's safe to precache that given URL.

An alternative approach would be to disallow passing in a string entirely, but I'm not sure that would be as... educational?... as this approach.

Reply:
And another alternative would be to *only* take in objects with both `url` and `revision` properties, and tell developers that they need to explicitly set `revision: null` if they're sure they're passing in a URL that is already revisioned. That's pretty draconian, though.